user/server-grpc-example.md

```diff
@@ -0,0 +1,156 @@
+structures and methods used for protocol buffer data. In the case of the Babelfish
+server protocol, you need to get [this `.proto`
+file](https://github.com/bblfsh/sdk/blob/master/protocol/generated.proto) from the
+SDK. Then you have to compile it to a source file with the required structures
+and methods. How you generate this module is language-dependent but usually it
+involves installing gRPC using your language package management tool (pip, cargo,
+npm, dub, etc) and then use one of the provided tools.
+
+For this example, we'll use the [GogoProtobuf
+implementation](https://github.com/gogo/protobuf) for the Go language. This
+provides the `protoc-gen-gogo` `.proto` compiler so we'll use it to generate the
+`.go` interface module:
+
+```bash
+$ protoc-gen-gogo --go_out=. generated.proto
+```
+
+This will create a generated.pb.go file that we could then directly import into
+out Go code. Since the SDK is also written in Go, you could skip this step and
+import the modules with the [already generated
+serializers](https://github.com/bblfsh/sdk/blob/master/protocol/generated.pb.go)
+in the Babelfish's SDK.
+
+## Making requests
+
+Now we'll write a simple program that sends a request to get UAST of a simple 
+"hello world" Python code that we'll provide. 
+
+```go
+package main
+
+import (
+    "fmt"
+    "os"
+    "time"
+    "context"
+
+    "google.golang.org/grpc"
+    "github.com/bblfsh/sdk/protocol"
+    "github.com/bblfsh/sdk/uast"
+)
+
+func main() {
+    // Connect to the running server
+    conn, err:= grpc.Dial("0.0.0.0:9432", grpc.WithTimeout(time.Second*2), 
+        grpc.WithInsecure())
+    if (err != nil) {
+        os.Exit(1)
+    }
+    client := protocol.NewProtocolServiceClient(conn)
+    req := &protocol.ParseUASTRequest{Filename: "hello.py",
+                                      Content:  "print('hello world!')",
+                                      Language: "python",}
+}
+```
+
+Now that we've created a request, we need to send it (previous code omitted):
+
+```go
+    resp, err := client.ParseUAST(context.TODO(), req)
+```
+
+
+## Reading and interpreting the response
+
+The code in the previous section returned a `ParseUASTResponse` object that will
+have the format of the [ParseUASTResponse](server-protocol.md#ParseUASTResponse)
+as seen on the [server protocol](server-protocol.md) page. You should check
+the `status` (`Status` in the case of Go, since public members start with
+uppercase); only a value of `protocol.Status.OK` will indicate sucess.
+
+The most important member of the `ParseUASTResponse` object is undoubtly
+`uast` (`UAST` in Go). This will contain a `Node` object which the [structure
+detailed in the previous page](server-protocol.md#Nodes). This first node
+returned would be the root node of the UAST, and you typically would iterate over
+the node children (contained in the aptly named `children` field) typically using
+the [visitor pattern](https://en.wikipedia.org/wiki/Visitor_pattern) and 
```

Comment:
Heh, I should problably have defined what I meant by _higher level_ better, certainly Python is not a language I'd include in. Anyway, the point is that _design patterns_ are highly opinionated and that's why it may make less sense to many people.

If you want to keep it, there's now  a _the_ that should be removed